core/internal/tensorboard/emitter.go

```diff
@@ -261,37 +253,42 @@ func (e *tfEmitter) EmitTable(
 	return nil
 }
 
-func (e *tfEmitter) EmitImage(
+func (e *tfEmitter) EmitImages(
 	key pathtree.TreePath,
-	img wbvalue.Image,
+	images []wbvalue.Image,
 ) error {
-	maybeRunFilePath, err := runRelativePath(
-		filepath.Join("media", "images"),
-		fmt.Sprintf(".%s", img.Format),
-	)
-	if err != nil {
-		return err
-	}
-	runRelativePath := *maybeRunFilePath
-	fsPath := filepath.Join(e.settings.GetFilesDir(), string(runRelativePath))
-
-	if _, err := os.Stat(fsPath); !os.IsNotExist(err) {
-		return fmt.Errorf("image file exists: %v", err)
+	// We assume that all images have the same format, height, and width.
+	// This aligns with the legacy way of logging images in TensorBoard.
+	format := images[0].Format
+	height := images[0].Height
+	width := images[0].Width
```

Comment:
Could you link out to docs or anything in the open source TB code that shows that we can rely on this? I try to do this throughout the package whenever we have assumptions like this. It would also be good practice to mention this requirement in the method's documentation and add a comment at call sites about why we believe the assumption holds.

Perhaps it would be even better to just check the assumption and return an error if it doesn't hold. To avoid cluttering this method, you could factor that out into a helper function.

```go
format := images[0].Format
width := images[0].Width
height := images[0].Height

if err := assertAllImagesMatch(
	images, format, width, height,
) {
	return err
}
```

(you could even make a helper that returns `(format string, width int, height int, err error)`, but it would be easier to unknowingly mix up the order of width and height)